### PR TITLE
run codegen against latest main to update licenses and api doc

### PR DESCRIPTION
**Pull request by Yongxuanzhang**
# Changes
This commit runs codegen against latest main to update licenses and api doc.

/kind misc

Signed-off-by: Yongxuan Zhang yongxuanzhang@google.com

# Submitter Checklist

As the author of this PR, please check off the items in this checklist:

- [ ] Has [Docs](https://github.com/tektoncd/community/blob/main/standards.md#docs) included if any changes are user facing
- [ ] Has [Tests](https://github.com/tektoncd/community/blob/main/standards.md#tests) included if any functionality added or changed
- [x] Follows the [commit message standard](https://github.com/tektoncd/community/blob/main/standards.md#commits)
- [x] Meets the [Tekton contributor standards](https://github.com/tektoncd/community/blob/main/standards.md) (including
  functionality, content, code)
- [x] Has a kind label. You can add one by adding a comment on this PR that contains `/kind <type>`. Valid types are bug, cleanup, design, documentation, feature, flake, misc, question, tep
- [ ] Release notes block below has been updated with any user facing changes (API changes, bug fixes, changes requiring upgrade notices or deprecation warnings)
- [ ] Release notes contains the string "action required" if the change requires additional action from users switching to the new release

# Release Notes

```release-note
NONE
```

### CODE REVIEW

File: docs/pipeline-api.md [hunk 1]
```diff
@@ -2,7 +2,7 @@
 ---
 title: Pipeline API
 linkTitle: Pipeline API
-weight: 404
+weight: 1000
```
Comment (afrittoli) [inline docs/pipeline-api.md:5]: Why this change here?

@geriom ^^^

**Comment by afrittoli** [comment]
I wonder how is it possible that this delta is not caught by CI, it sounds like something is broken? @vdemeester wdyt?

**Comment by Yongxuanzhang** [comment]
It may make sense to check the pipeline-api file. But I'm not sure about the licenses files

**Comment by JeromeJu** [comment]
Wondering https://github.com/tektoncd/pipeline/pull/6037 is not a blocker any more?

**Comment by vdemeester** [comment]
> I wonder how is it possible that this delta is not caught by CI, it sounds like something is broken? @vdemeester wdyt?

Good question 🤔

File: docs/pipeline-api.md (hunk 1 above)
Comment (geriom) [reply docs/pipeline-api.md:5]: Definitely we should keep this as 404 (or anything other weight in the 400s).

Context: https://github.com/tektoncd/pipeline/pull/6117

**Review by geriom — CHANGES_REQUESTED** [review]

**Comment by Yongxuanzhang** [comment]
> Signed-off-by: Yongxuan Zhang [yongxuanzhang@google.com](mailto:yongxuanzhang@google.com)

It is still a blocker. 😂  I manually removed them. Will try to get that merged as well

File: docs/pipeline-api.md (hunk 1 above)
Comment (Yongxuanzhang) [reply docs/pipeline-api.md:5]: This is autogenerated, should I fix it manually?

Comment (Yongxuanzhang) [reply docs/pipeline-api.md]: I think we may want to update the weight here
https://github.com/tektoncd/pipeline/blob/0b699455199cf1b80212ef1becade922883b0186/hack/update-reference-docs.sh#L26
@geriom

**Review by lbernick — APPROVED** [review]
OK to merge but +1 on fixing CI so that this is checked

**Comment by dibyom** [comment]
/lgtm

Opened #6210

**Comment by Yongxuanzhang** [comment]
that's weird, why the build fails but passed before?

**Comment by JeromeJu** [comment]
> that's weird, why the build fails but passed before?

https://github.com/tektoncd/plumbing/commit/50366b36bac7ed4c59ed0f454a5d629db230693e got in earlier. I guess we should not need to `register.go` anymore
